plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/controls/querylog/QueryLogViewer.java

```diff
@@ -142,6 +142,15 @@ private static class ColumnDescriptor {
 
         @Override
         String getText(QMEvent event, boolean briefInfo) {
+            if (event.getObject() instanceof QMMConnectionInfo) {
+                switch (event.getAction()) {
+                    case END:
+                        return timeFormat.format(event.getObject().getCloseTime());
+                    case BEGIN:
+                    default:
+                        timeFormat.format(event.getObject().getOpenTime());
+                }
+            }
```

Comment:
I don't think this `switch` statement is required here; moreover, the expression in the `default` branch is never returned but works because the last return produces the same result.

```suggestion
            if (event.getObject() instanceof QMMConnectionInfo && event.getAction() == QMEventAction.END) {
                return timeFormat.format(event.getObject().getCloseTime());
            }
```

Lastly, I'm not sure if the check for `QMMConnectionInfo` is needed as all QM objects have "open" and "close" times.

Reply:
We do need to check for QMMConnectionInfo  as we want to do it only for disconnect from database events.

Reply:
Indeed, but what about other events?